src/lib/defaultCommands.js

```diff
@@ -442,10 +442,33 @@ var defaultCmds = {
 	// START_COMMAND: Code
 	code: {
 		exec: function () {
-			this.wysiwygEditorInsertHtml(
-				'<code>',
-				(IE_BR_FIX ? '' : '<br />') + '</code>'
-			);
+			var editor = this;
+			var range = editor.getRangeHelper().selectedRange();
+			var selected = editor.getRangeHelper().selectedHtml();
+			var wholeLine = false;
+
+			// Check if the whole line was selected
+			// It could be the whole text of the text type node (3)
+			// or innerHTML of an element node (1) if there were some markups
+			// in the text, so its containted in a paragraph <p>...</p>
+			if ((range.commonAncestorContainer.nodeType === 3 &&
+				range.commonAncestorContainer.wholeText === selected) ||
+				(range.commonAncestorContainer.nodeType === 1 &&
+				range.commonAncestorContainer.innerHTML === selected)) {
+				wholeLine = true;
+			}
+
+			if (editor.opts.allowInlineCode && !wholeLine && selected &&
+				selected.length > 0 && !selected.includes('<p>')) {
+				this.wysiwygEditorInsertHtml(
+					'<span class="inline-code" data-inline-code="1">', '</span>'
+				);
+			} else {
+				this.wysiwygEditorInsertHtml(
+					'<code>',
+					(IE_BR_FIX ? '' : '<br />') + '</code>'
+				);
+			}
```

Comment:
This new doesn't work and it is not as easy as it looks. Better leave that to two buttons.
A user may want to transform a single line of code into a code block.
E.g. How do you manage a press of `Enter` while inside the inline-code block when the user had used the code block? I can't find such code.

Reply:
I meant that the concept doesn't work, not the code itself. (the reason is in the prev comment)

Reply:
The code marks the selected text as code. I can't see any inconsistence.
Play with the code button here on GitHub or phpBB forum and you will see exactly the same behavior.
Unfortunately I couldn't find CKEditor demo with code button, so I don't know how they do that.

The proposed behavior was very intuitive for me, while editing about 100 pages with code snippets all over. And consistent with GitHub and phpBB behavior as I wrote...

Anyway, don't integrate the PR yet, I want eliminate the use of <span> and move to <code> for inline and <pre><code> for code blocks, it will address issue #648 as well and the implementation will probably be better.
I'll send it as a separate PR.

Reply:
the `[c]` and `[code]` bbcodes in phpBB central forum are handled as different BBCodes and they are always clearly different.
Here, you are joining the inline code with the block code as a single command. That is my problem with this. If you make a new command for the inline code, then I have no issues with this.

Reply:
Sorry, but I think we are talking about something different.

The code introduce `[c]` bbcode in addition to `[code]` that is already implemented.
The code decides whether to use `[c]` or `[code]` based on the selected text:
full line selected - use `[code]`
text in a line - use `[c]`

Currently the `[code]` block is marked with `<code>` HTML tag, while `[c]` tag is marked with `<span class="inline-code">` in non-source view.

`[c]` and `[code]` are absolutely different codes, so I don't understand your rejections...

Reply:
I'll try like this:
How do you show a user the difference between both outcomes?
When a user presses `Enter`, the outcome isn't the same depending on both possible situations.
![enter_on_inline_code](https://user-images.githubusercontent.com/639467/38458101-e35f8cc0-3a91-11e8-90d7-581e8989dee1.gif)
Because, if the new line is already there, the style is removed and reapplied....
![if_unstyled_and_restyled](https://user-images.githubusercontent.com/639467/38458124-39478e4e-3a92-11e8-8567-560013454be9.gif)
The outcome doesn't fit with the previous state.

Reply:
I will check why your last example didn't choose the old code block style, but I'm ok with the rest of the behaviour...